common/date.go

```diff
@@ -0,0 +1,43 @@
+package common
+
+import (
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ParseDate parses a date string of the format YYYY-MM-DD
+func ParseDate(date string) (t time.Time, err error) {
+	layout := "2006-01-02" //this represents YYYY-MM-DD
```

Comment:
This function is used outside of the ParseDateRange function too, so it should probably exist. We can make layout a constant but I don't think it makes a difference, your call given that this function needs to exist either way